### ISSUE TITLE
Add detailed documentation to engine nacelles

### ISSUE THREAD

**Issue by joergbrech** [DLR-SL/CPACS#731]
The engine nacelles have been around for a while now, but the documentation only contains an image of a nacelle without a description how to create one with CPACS. We should add more details here.  We have the schema, the implementation in TiGL and some CPACS files to start.

**Comment by MarAlder** [comment]
I added the documentation and an example file:

### Documentation
- [x] better descriptions
- [x] [engineNacelleType](https://www.cpacs.de/documentation/CPACS_development/html/268c6e45-e05b-ef0d-86cf-76dee950966b.htm)
- [x] [rotationCurveType](https://www.cpacs.de/documentation/CPACS_development/html/bb90c845-61b3-903f-9fb7-54bf6fcbef00.htm)
- [x] [nacelleSectionType](https://www.cpacs.de/documentation/CPACS_development/html/ec73cb6b-57dc-0814-0880-5aae93978380.htm)
- [x] [nacelleGuideCurveType](https://www.cpacs.de/documentation/CPACS_development/html/039183bc-03cd-b114-19e6-79dfe50b13db.htm)
- [x] [nacelleCenterCowlType](https://www.cpacs.de/documentation/CPACS_development/html/6d17b3b0-8d6d-8049-6fe2-1897d177ae47.htm)

### Example file
- [nacelles.xml](https://github.com/DLR-SL/CPACS/blob/develop/examples/nacelles.xml)
![nacelles](https://user-images.githubusercontent.com/43143741/140908442-6d81ce08-0000-4c16-b883-a08d33aeaf6b.png)

## Open questions

- [ ] In case of three-dimensional fan blades: is the coordinate center the most forwared (in negative streamwise direction) postion or its geometric center?
![nacelle](https://user-images.githubusercontent.com/43143741/140911127-5cc2edea-c0f3-43a9-9762-1c4421879b17.png)

## Derived issues

This does not belong to the present issue, but I want to raise some potential issues which could be discussed in future if necessary:
- The center cowl has no x-scaling. Maybe it would make sense to add it so that the profile can be predefined between 0..1
- There is no overall `transformation` node. Could it be useful?
- Maybe it makes sense to set the `rotationCurve` optional?
- guide curves `description` element should be optional

## Review

- [ ] @codingpoets: I think you were involved in developing the nacelles, right? Could you please verify that the documentation correctly reflects the intended approach?
- [ ] @joergbrech: Can you check the consistency between the descriptions and the TiGL implementation?
- [ ] @cliersch: Can you check with Patrick whether the descriptions answer the questions you had a few weeks ago?
- [ ] @swoehler: You have already gained some experience with the nacelles. Could you check if the docs are consistent with your understanding/your openAD implementation?

**Comment by joergbrech** [comment]
Nice! Thanks for the great work @MarAlder!

Regarding your mentioned issues I have no strong opinions.
 - We discussed wether the rotation curve should be optional or not, and I think the consensus back than was that it is impossible to build an engine nacelle without a rotationally symmetric interior.
 - I also think to remember that we discussed wether we should add x-scaling to the center cowl, I cannot remember why we descided not to do it. It is possible that we just postponed the decision: We wanted to have a working prototype first.

Finally, the nacelle guide curve profiles need some further documentation, or at least I couldn't find it in your links. The guide curve points are interpreted as (r and x) offsets from a cubic polynomial:

![image](https://user-images.githubusercontent.com/25240961/141330738-f33bedba-68b0-4232-864c-ef6ccfef08e8.png)

This is a bit ugly, but the intuition behind it is that we needed a baseline for the guide curves that works between sections that are positioned at different radial positions. So the baseline is a smooth transition from the radius of the start section to the end section, which is given by the above cubic polynomial.
